### ISSUE TITLE
Interests system

### ISSUE THREAD

**Issue by ramshorst** [hckrs/hckrs.io#51]
I have been talking to @bhabegger this afternoon about what the good techniques are to use the interest data we will collect. He wanted to know if we have SSH access on our Heroku account, so that we can use Elastic search for example. I figured that this would not be a good idea on the small computing power we have at Heroku, but maybe it makes sense to use a AWS account for this? @bhabegger also told me that universities often have access to free accounts, is this maybe the case of you university @Jarnoleconte ?

As a data source he suggested we use the Stack Overflow tags. It turns ou that all the data is available for download : https://archive.org/details/stackexchange, the one interesting us first is the list with all the 10.000 tags and their occurence : https://archive.org/download/stackexchange/stackoverflow.com-Tags.7z
We could make a relational database by figuring out what tags are used together in posts, or are linked to the same users. Stack exchange does this pretty well : an example of the results displayed on the NodeJs tag page:
![screen shot 2014-10-17 at 15 05 36](https://cloud.githubusercontent.com/assets/5200239/4679148/78400a80-55fe-11e4-8284-b7b6088692e8.png)
http://stackoverflow.com/tags/node.js/info

Sorry @Mellowlicious that there is still no design ; it takes me more time to figure out what to build than i would like to, but the good news is that we won't build something useless ;)

**Comment by Mellowlicious** [comment]
Yesterday I finally took the time to write a program to parse the stackoverflow datadump. This means that I went through all opening posts, extracted the tags, and marked how many times the same tags appeared next to each other in the same post. So for example, if the tags are <c#><.net><lists> then there would be 1 relation between c# and .net, one relation between c# and lists, and one relation between .net and lists.

Note that this is all assuming my program works correctly.

Around 57% of the relations only occur once
Only 1.1% of the relations occur 100 times or more

Tomorrow I will double check my program to make sure there are no errors, then fill in a small development grapheneDB to start testing out suggestions.
